utils/fileUploads.js

```
@@ -0,0 +1,134 @@
+  if (req.method !== "POST") {
+    return res.status(405).json({ message: "Only POST requests are accepted!", code: 405 });
+  }
+
+  if (!req.headers["content-type"] || !req.headers["content-type"].includes("multipart/form-data")) {
+    return res.status(400).json({ message: "content-type must be multipart/form-data", code: 400 });
+  }
+
+  const Connect_ID = req.query.Connect_ID;
+  const siteAcronymLower = req.query.siteAcronym?.toLowerCase();
+
+  if (!Connect_ID || !siteAcronymLower) {
+    return res.status(400).json({ message: "Missing Connect_ID or siteAcronym in query parameters", code: 400 });
+  }
+
+  const bucketName = `${pathologyReports}_${siteAcronymLower}_${tierStr}`;
+  const busboy = Busboy({ headers: req.headers });
+  const uploadPromises = [];
+  let fields = {};
+  let bucket = null;
+
+  busboy.on("field", (fieldname, val) => {
+    fields[fieldname] = val;
+  });
+
+  busboy.on("file", async (fieldname, file, { filename, mimetype }) => {
+    if (!bucket) {
+      const [exists] = await storage.bucket(bucketName).exists();
+      if (!exists) {
+        await storage.createBucket(bucketName, {
+          location: "US",
+          project: storage.projectId,
+        });
+        await awaitNewBucketReady(bucketName);
+      }
+      bucket = storage.bucket(bucketName);
+    }
+
+    const filePath = `${Connect_ID}/${filename}`;
+    const fileUpload = bucket.file(filePath);
+    const stream = file.pipe(
+      fileUpload.createWriteStream({
+        metadata: { contentType: mimetype },
+      })
+    );
+
+    uploadPromises.push(
+      new Promise((resolve, reject) => {
+        stream.on("finish", () => resolve(filename));
+        stream.on("error", reject);
+      })
+    );
+  });
+
+  busboy.on("finish", async () => {
+    try {
+      const promiseResults = await Promise.allSettled(uploadPromises);
+      const successFilenames = promiseResults
+        .filter((result) => result.status === "fulfilled")
```

This seems like it will swallow errors if some of the files upload but some of them don't. Should this be recorded somewhere?

---

Is there a straightforward way to compare the initial file list with `successFilenames` so we know which ones failed in the event of service issues and or possibly return/log `206 partial` like we do in the API when some items fail? Or does this mix current and previously updated data?

---

It's easy to get failed filenames and return them, if errors occur. 206 might be the most appropriate code when mixed cases happen.
I'll do it shortly.